src/AWS.Deploy.Orchestration/DeploymentBundleHandler.cs

```diff
@@ -74,9 +74,11 @@ public async Task BuildDockerImage(CloudApplication cloudApplication, Recommenda
             DockerUtilities.TryGetAbsoluteDockerfile(recommendation, _fileManager, _directoryManager, out var dockerFile);
 
             var dockerBuildCommand = $"docker build -t {imageTag} -f \"{dockerFile}\"{buildArgs} .";
-            if (RuntimeInformation.OSArchitecture != Architecture.X64)
+            var currentArchitecture = RuntimeInformation.OSArchitecture == Architecture.Arm64 ? SupportedArchitecture.Arm64 : SupportedArchitecture.X86_64;
+            var dockerPlatform = recommendation.DeploymentBundle.EnvironmentArchitecture == SupportedArchitecture.Arm64 ? "linux/arm64" : "linux/amd64";
+            if (currentArchitecture != recommendation.DeploymentBundle.EnvironmentArchitecture)
```

Comment:
`            if (currentArchitecture != recommendation.DeploymentBundle.EnvironmentArchitecture)` what exactly is this check for? Im trying to understand the context of what its actually doing.